mne/viz/_brain/_brain.py

```diff
@@ -2368,6 +2372,76 @@ def add_foci(self, coords, coords_as_verts=False, map_surface=None,
                                   opacity=alpha, resolution=resolution)
             self._renderer.set_camera(**views_dicts[hemi][v])
 
+    @fill_doc
+    def add_sensors(self, info, trans=None, ch_types=None, colors=None,
+                    scales=None, alphas=None):
+        """Add an ROI label to the image.
+
+        Parameters
+        ----------
+        info : mne.Info
+            The measurement information with sensor locations.
+        %(trans)s
+        ch_types : dict
+            Channel type keyword arguments to be passed to
+            :func:`mne.pick_types`.
+        colors : str | dict
+            The color of the sensors. If dict, channel names or types can be
+            keys with values of matplotlib colors. Default is all sensors
+            yellow.
+        scales : float | dict
+            The size of the sensors. If dict, channel names or types can be
+            keys with values for the size in mm. Default is all sensors 5 mm.
+        alphas : float | dict
+            The opacity of the sensors or, if dict, each sensor by name or
+            type. Default is all sensors 0.5 opacity.
+
+        Notes
+        -----
+        .. versionadded:: 0.24
+        """
+        from matplotlib.colors import colorConverter
+        if ch_types is not None:
+            picks = pick_types(info, **ch_types)
+            info = info.copy().pick_channels(
+                [info.ch_names[idx] for idx in picks])
+        ch_coords = np.array([ch['loc'][:3] for ch in info['chs']])
+        for idx, ch_coord in enumerate(ch_coords):
+            # coordinate frame/location info
+            coord_frame = info['chs'][idx]['coord_frame']
+            if coord_frame == FIFF.FIFFV_COORD_DEVICE:
+                ch_coord = apply_trans(info['dev_head_t']['trans'], ch_coord)
+                coord_frame = FIFF.FIFFV_COORD_HEAD  # now it's in head
+            if coord_frame == FIFF.FIFFV_COORD_HEAD:
+                if trans is None:
+                    raise ValueError('`trans` required for channel data in '
+                                     'the "meg" or head" coordinate frames')
+                trans = _ensure_trans(trans, 'head', 'mri')
+                ch_coord = apply_trans(trans['trans'], ch_coord) * 1000
+                coord_frame = FIFF.FIFFV_COORD_MRI  # now in surface RAS
+            if coord_frame != FIFF.FIFFV_COORD_MRI:
+                raise RuntimeError(
+                    '`info` coordinate frame must be "meg", "head" or "mri", '
+                    f'got {coord_frame} for {info.ch_names[idx]}. You may '
+                    'want to use `ch_types=dict(meg=True, eeg=True)` '
+                    'for instance')
```

Comment:
Same comment here, a lot of work has gone into `plot_alignment` to do coordinate frame gymnastics, so code should be deduplicated / refactored to be shared between the functions

Reply:
So there's a lot more gymnastics in `plot_alignment` because the plot can be in device, head or mri coordinates. It makes sense to me that everything should just be in mri/surface RAS but that would be a depreciation. What do you think, make `mne.viz.Brain` support coordinate frame shifts or deprecate the coordinate frame support in `plot_alignment` or a third option?

Reply:
> It makes sense to me that everything should just be in mri/surface RAS but that would be a depreciation

Not always, this is why we provide the option. As a rule in MNE we try to only provide options that people need, and it's useful to be able to align to each coordinate frame (esp. for example in combined M-EEG recordings where you can look at alignment with no MRI whatsoever!).

> make mne.viz.Brain support coordinate frame

This can/should just stay MRI-only since it's brain/recon-centric

> a third option?

Yes:

1. Make a private helper function `_convert_sensors_to(info, ..., coord_frame)` that converts sensor coordinates from whatever coordinate frame they're natively in to whatever coordinate frame they need to be in for plotting. (This code exists already in `plot_alignment` and "just" needs to be pulled out.) 
2. Have plot_alignment call this, with `coord_frame=coord_frame`
3. Have `Brain` call this with `frame_to='mri'`

This kind of kills the dream of having `plot_alignment` be a thin wrapper to Brain, but I'm okay with that. It's 98% as readable and maintainable if both just call a set of shared private functions like this, and we keep an optimal user-facing API in terms of coordinate frames (i.e., allow option to change it in `plot_alignment` where it is useful, always use MRI for Brain since that's brain/recon-centric).